src/main/java/org/jenkins_ci/plugins/flexible_publish/ConditionalPublisher.java

```diff
@@ -169,20 +170,41 @@ public ConditionalPublisherDescriptor getDescriptor() {
     }
 
     public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener listener) {
-        for (BuildStep publisher: getPublisherList()) {
-            if (!runner.prebuild(condition, publisher, build, listener)) {
-                return false;
-            }
+        if (getPublisherList().size() <= 0) {
+            return true;
+        } else if (getPublisherList().size() > 1) {
+            return runner.prebuild(condition, new RunAllBuilder(getPublisherList()), build, listener);
         }
-        return true;
+        return runner.prebuild(condition, getPublisherList().get(0), build, listener);
     }
 
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
                                                                                                 throws InterruptedException, IOException {
-        for (BuildStep publisher: getPublisherList()) {
-            if (!runner.perform(condition, publisher, build, launcher, listener)) {
+        if (getPublisherList().size() <= 0) {
+            return true;
+        } else if (getPublisherList().size() > 1) {
+            return runner.perform(condition, new RunAllBuilder(getPublisherList()), build, launcher, listener);
```

Comment:
@oleg-nenashev 
Thanks for your comment. I'm glad for your help.
I agree that's the reasonable way to resolve the conflict.

I plan a following feature. Would you review this?:
* Introduces "Execution strategry" in the advanced section for each conditional action (that is, for each condition).

|Execution strategy|Behavior                                                                |Default in <= 1.14.1      |Default in >= 1.15|
|:----------------:|:-----------------------------------------------------------------------|:------------------------:|:----------------:|
|Fail at end       |Run all publishers even when some of them fail                          |                          |x                 |
|Fail fast         |Stop run publishers in that condition immediately when some of them fail|x (backward compatibility)|                  |

* Example:
    * This results "Some publisher" performed:
    ```
    Conditional Action
        Run?: Always
        Action:
            Failure publisher
            Some publisher
        Advanced:
            Execution strategy: Fail at end
    ```
    * This results "Some publisher" *NOT* performed:
    ```
    Conditional Action
        Run?: Always
        Action:
            Failure publisher
            Some publisher
        Advanced:
            Execution strategy: Fail fast
    ```

* Notes
    * This covers the backward compatibility for the behavior with failures.
        * I suppose it isn't the point whether the condition is evaluated only once or each time for each action.
    * This feature will work only inside each conditional actions. Flexible publish runs all conditions even some of them fail.
        * Example: This results "Some publisher 1" *NOT* performed, but "Some publisher 2" performed:
        ```
        Conditional Action
            Run?: Always
            Action:
                Failure publisher
                Some publisher 1
            Advanced:
                Execution strategy: Fail fast
        Conditional Action
            Run?: Always
            Action:
                Some publisher 2
            Advanced:
                Execution strategy: Fail fast
        ```
    * Migration from <= 1.14.1 is performed in readResolve.
    * I'll implement that by switching `RunAllBuilder` and `FailFastBuilder` (will be added).